services/s3/src/test/java/software/amazon/awssdk/services/s3/S3EndpointResolutionTest.java

```diff
@@ -508,4 +509,62 @@ private S3Configuration withDualstackAndAccelerateEnabled() {
                               .build();
     }
 
+
+    @Test
+    public void accessPointArn_us_east_1_region_aws_global_useArnRegionTrue() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.AWS_GLOBAL)
+                .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build()).build();
+        String accessPointArn = "arn:aws:s3:us-east-1:12345678910:accesspoint:foobar";
+        s3Client.getObject(GetObjectRequest.builder().bucket(accessPointArn).key("someKey").build());
+        assertThat(mockHttpClient.getLastRequest().getUri().getHost())
+                .isEqualTo("foobar-12345678910.s3-accesspoint.us-east-1.amazonaws.com");
+
+    }
+
+    @Test
+    public void accessPointArn_us_east_1_region_aws_global_useArnRegionFalse_throwsIllegalArgumentException() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.AWS_GLOBAL).build();
+        String accessPointArn = "arn:aws:s3:us-east-1:12345678910:accesspoint:foobar";
+        assertThatThrownBy(() -> s3Client.listObjects(ListObjectsRequest.builder().bucket(accessPointArn).build()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("region");
+    }
+
+    @Test
+    public void accessPointArn_us_east_1_region_s3_external_1_useArnRegionFalse_throwsIllegalArgumentException() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.of("s3-external-1")).build();
+        String accessPointArn = "arn:aws:s3:us-east-1:12345678910:accesspoint:foobar";
+        assertThatThrownBy(() -> s3Client.listObjects(ListObjectsRequest.builder().bucket(accessPointArn).build()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("region");
+    }
+
+    @Test
+    public void accessPointArn_us_gov_west_1_clientRegion_fips_us_gov_east_1_useArnRegionTrue() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.of("fips-us-gov-east-1"))
+                .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build()).build();
+        String accessPointArn = "arn:aws-us-gov:s3:us-gov-west-1:123456789012:accesspoint:myendpoint";
+
+        assertThatThrownBy(() -> s3Client.getObject(GetObjectRequest.builder().bucket(accessPointArn).key("someKey").build()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("region");
```

Comment:
useArnRegionEnabled is true for here. What's the error message that's thrown? Is it still the "enable use-arn-region" one?

Reply:
Adding separate error message for fips case